nipype/utils/tests/test_filemanip.py

```diff
@@ -670,3 +671,26 @@ def test_write_rst_list(tmp_path, items, expected):
     else:
         with pytest.raises(expected):
             write_rst_list(items)
+
+
+def nfs_unlink(pathlike, *, dir_fd=None):
+    if dir_fd is None:
+        path = Path(pathlike)
+        deleted = path.with_name(".nfs00000000")
+        path.rename(deleted)
+    else:
+        os.rename(pathlike, ".nfs1111111111", src_dir_fd=dir_fd, dst_dir_fd=dir_fd)
```

Comment:
I do like your idea of avoiding branching. What if we just construct the target correctly?

```suggestion
    target = os.path.join(os.path.dirname(pathlike), ".nfs00000000")
    os.rename(pathlike, target, src_dir_fd=dir_fd, dst_dir_fd=dir_fd)
```

Demo:

```Python
>>> [os.path.join(os.path.dirname(pathlike), ".nfs00000000")
...  for pathlike in ["path/subdir/busyfile", "busyfile"]]
['path/subdir/.nfs00000000', '.nfs00000000']
```